ibis/backends/base/sql/alchemy/__init__.py

```diff
@@ -91,7 +94,8 @@ def _create_table_as(element, compiler, **kw):
     if element.temp:
         stmt += "TEMPORARY "
 
-    return stmt + f"TABLE {element.name} AS {compiler.process(element.query, **kw)}"
+    name = compiler.preparer.quote(quoted_name(element.name, quote=element.quote))
```

Comment:
Is it necessary to call `quote` on `quoted_name`?
If it is, maybe a brief comment on what's going on.